Assets/MRTK/Core/Inspectors/MixedRealityToolboxWindow.cs

```diff
@@ -0,0 +1,268 @@
+                    {
+                        RenderSection(bucket);
+                    }
+                }
+            }
+        }
+
+        private void RenderSection(ToolboxCategory bucket)
+        {
+            using (new EditorGUILayout.VerticalScope(EditorStyles.helpBox))
+            {
+                string key = $"MixedRealityToolboxWindow_{bucket.CategoryName}";
+                if (InspectorUIUtility.DrawSectionFoldoutWithKey(bucket.CategoryName, key, MixedRealityStylesUtility.BoldTitleFoldoutStyle))
+                {
+                    InspectorUIUtility.DrawDivider();
+                    EditorGUILayout.Space();
+
+                    bool isCategoryNameSearchMatch = IsSearchMatch(bucket.CategoryName, searchString);
+
+                    List<ToolboxItem> validItems = new List<ToolboxItem>();
+                    foreach (var item in bucket.Items)
+                    {
+                        if (item != null && item.Prefab != null 
+                            && (isCategoryNameSearchMatch || IsSearchMatch(item, searchString)))
+                        {
+                            validItems.Add(item);
+                        }
+                    }
+
+                    // Render grid of toolbox items
+                    int itemsPerRow = Mathf.Max((int)(position.width / ToolboxItemWidth), 1);
+
+                    for (int row = 0; row <= validItems.Count / itemsPerRow; row++)
+                    {
+                        using (new EditorGUILayout.HorizontalScope())
+                        {
+                            int startIndex = row * itemsPerRow;
+                            for (int col = 0; col < itemsPerRow && startIndex + col < validItems.Count; col++)
+                            {
+                                var item = validItems[startIndex + col];
+                                RenderToolboxItem(item);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private void RenderToolboxItem(ToolboxItem item)
+        {
+            if (item == null || item.Prefab == null)
+            {
+                return;
+            }
+
+            var buttonContent = new GUIContent()
+            {
+                image = item.Icon,
+                text = string.Empty,
+                tooltip = item.DocURL,
+            };
+
+            using (new EditorGUILayout.VerticalScope())
+            {
+                using (new EditorGUILayout.HorizontalScope())
+                {
+                    GUILayout.FlexibleSpace();
+                    using (new EditorGUI.DisabledGroupScope(false))
+                    {
+                        if (GUILayout.Button(buttonContent, GUILayout.MaxHeight(ToolboxItemHeight), GUILayout.Width(ToolboxItemButtonWidth)))
+                        {
+                            Selection.activeObject = Instantiate(item.Prefab);
+                        }
```

Comment:
We may need to add a guard for the Unity UI elements.  This will make sure to only add the Unity UI element to the scene if an MRTK canvas exists.  We can also add a MRTK canvas and the button together if a canvas does not exist.

Reply:
Ooh good call. Do you have an idea of what this might look like? Search the scene for an MRTK canvas or something?

Reply:
Yep!
When a regular canvas is added to the scene, you can convert it to an MRTK canvas, which adds the CanvasUtility to the game object

![image](https://user-images.githubusercontent.com/53493796/80262478-65141580-8642-11ea-9eaa-769d68884ffa.png)

So maybe we could search for a canvas with the canvas utility attached and if it exists then add the UnityUI element as a child of the canvas object

![image](https://user-images.githubusercontent.com/53493796/80262548-9b519500-8642-11ea-9767-d5e96b205935.png)

Reply:
Added a dropdown to the Unity UI section, which finds all `CanvasUtility`s in the scene (and thus valid MRTK canvases):

![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/3580640/80771057-026dbe80-8b07-11ea-86b6-74b36e3e45d8.png)

When a component's button is selected, it's automatically placed under the selected canvas in the dropdown.

If none are detected, the buttons are disabled and the following shows up:

![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/3580640/80771091-24ffd780-8b07-11ea-844c-c84e5b599bd3.png)